stdlib/public/Backtracing/BacktraceFormatter.swift

```diff
@@ -0,0 +1,983 @@
+                                    indent theIndent: Int = 2) -> String? {
+    guard let fp = fopen(sourceLocation.path, "rt") else { return nil }
+    defer {
+      fclose(fp)
+    }
+
+    let indent = String(repeating: " ", count: theIndent)
+    var lines: [String] = []
+    var line = 1
+    let buffer = UnsafeMutableBufferPointer<CChar>.allocate(capacity: 4096)
+    var currentLine = ""
+
+    let maxLine = sourceLocation.line + options._sourceContextLines
+    let maxLineWidth = max("\(maxLine)".count, 4)
+
+    let doLine = { sourceLine in
+      if line >= sourceLocation.line - options._sourceContextLines
+           && line <= sourceLocation.line + options._sourceContextLines {
+        let untabified = untabify(sourceLine)
+        let code = options._theme.code(untabified)
+        let lineNumber = options._theme.lineNumber(pad("\(line)",
+                                                       to: maxLineWidth,
+                                                       aligned: .right))
+        let theLine: String
+        if line == sourceLocation.line {
+          let highlightWidth = options._width - 2 * theIndent
+          theLine = options._theme.crashedLine(pad("\(lineNumber)│ \(code) ",
+                                                   to: highlightWidth))
+        } else {
+          theLine = "\(lineNumber)│ \(code)"
+        }
+        lines.append("\(indent)\(theLine)")
+
+        if line == sourceLocation.line {
+          // sourceLocation.column is an index in UTF-8 code units in
+          // `untabified`.  We should point at the grapheme cluster that
+          // contains that UTF-8 index.
+          let adjustedColumn = max(sourceLocation.column, 1)
+          let utf8Ndx
+            = untabified.utf8.index(untabified.utf8.startIndex,
+                                    offsetBy: adjustedColumn,
+                                    limitedBy: untabified.utf8.endIndex)
+            ?? untabified.utf8.endIndex
+
+          // Adjust it to point at a grapheme cluster start
+          let strNdx = untabified.index(
+            untabified.index(utf8Ndx, offsetBy: 1,
+                             limitedBy: untabified.endIndex)
+              ?? untabified.endIndex,
+            offsetBy: -1,
+            limitedBy: untabified.startIndex) ?? untabified.startIndex
+
+          // Work out the terminal width up to that point
+          let terminalWidth = measure(untabified.prefix(upTo: strNdx))
+
+          let pad = String(repeating: " ",
+                           count: max(terminalWidth - 1, 0))
+
+          let marker = options._theme.crashLocation("▲")
+          let blankForNumber = String(repeating: " ", count: maxLineWidth)
+
+          lines.append("\(indent)\(blankForNumber)│ \(pad)\(marker)")
+        }
+      }
+    }
+
+    while feof(fp) == 0 && ferror(fp) == 0 {
+      guard let result = fgets(buffer.baseAddress,
+                               Int32(buffer.count), fp) else {
```

Comment:
I think `CInt` is more appropriate than `Int32` here (and other places too)

Reply:
Yeah, I think this is a result of me leaning on the compiler's fixit messages to catch these. I'll do a pass through and see if I can pick them all up.

Reply:
I fixed this one, but I wasn't sure where else you had in mind.